revive.toml

```diff
@@ -0,0 +1,104 @@
+# config for https://github.com/mgechev/revive
+#
+# largely the same reasoning as https://github.com/uber/cadence , but not intended to be identical.
+# libraries have different needs / goals than applications, so some drift is desirable.
+ignoreGeneratedHeader = false
+severity = "error"
+confidence = 0.8
+errorCode = 1
+warningCode = 0
+
+[directive.specify-disable-reason]
+    severity = "error"
+
+#### roughly what golint does.  probably only disable noisy ones.
+
+[rule.blank-imports]
+[rule.context-as-argument]
+[rule.context-keys-type]
+[rule.dot-imports]
+[rule.error-naming]
+[rule.error-return]
+[rule.error-strings]
+[rule.errorf]
+# [rule.exported] # disabled due to lack of value / encouraging bad habits.  it complains about internal packages.
+[rule.if-return]
+[rule.increment-decrement]
+[rule.indent-error-flow]
+# [rule.package-comments] # disabled due to lack of value / encouraging bad habits.  it complains about internal packages.
```

Comment:
nit, comment refers to package exports, I think you meant requiring comments on packages

Reply:
both complain about internals, e.g. package-comments has this:
```
internal/common/backoff/retry.go
  (21, 1)  https://revive.run/r#package-comments  should have a package comment
```

I'd be game to enable package-comments tbh, I think they have frequent-enough value even in internals.  There's just like 15 of these at the moment and I don't think we particularly care.